### PR TITLE
Document suggested rust-analyzer configuration

### PR DESCRIPTION
**Pull request by divergentdave**
This adds a section to `CONTRIBUTING.md` on how to best use rust-analyzer in this repository.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,6 +72,23 @@ There have not been that many people familiar with DNS internals, networking, se
...
+    "rust-analyzer.linkedProjects": [
+        "/path/to/hickory-dns/Cargo.toml",
+        "/path/to/hickory-dns/conformance/Cargo.toml",
+        "/path/to/hickory-dns/fuzz/Cargo.toml",
+        "/path/to/hickory-dns/tests/e2e-tests/Cargo.toml"
+    ],
+    "rust-analyzer.cargo.features": "all"
+    // etc.
+}
+```
```
Comment (djc) [inline CONTRIBUTING.md:90]: Can you think of a reason not to just commit this into the repo?

Comment (divergentdave) [reply CONTRIBUTING.md:90]: I haven't been able to get it to work with relative paths in the past

**Review by djc — APPROVED** [review]
This makes sense to me. @marcus0x62 any thoughts?

**Comment by marcus0x62** [comment]
> This makes sense to me. @marcus0x62 any thoughts?

I don't really have an opinion on this - I use rustic in emacs, and rust-analyzer pretty much just works there.

**Pull request merged**